### PR TITLE
Children array error

### PR DESCRIPTION
**Pull request by alexcjohnson**
Trying to give a more helpful error message than users encounter today, like #1024, when forgetting to wrap `children` in an array inside components. Obviously if we ever do Components as Props that would need to be taken into account here - but I suspect that would be fine, each component would need to register internally which props can accept components.

- [x] I have added tests, or extended existing tests, to cover any new features or bugs fixed in this PR

### optionals

- [x] I have added entry in the `CHANGELOG.md`

### CODE REVIEW

File: dash/development/base_component.py [hunk 1]
```diff
@@ -94,6 +94,14 @@ def __init__(self, **kwargs):
                         ", ".join(sorted(self._prop_names))
                     )
                 )
+
+            if k != "children" and isinstance(v, Component):
+                raise TypeError(
+                    "Component detected as a prop other than `children`\n" +
+                    "Did you forget to wrap multiple `children` in an array?" +
```
Comment (Marc-Andre-Rivet) [inline dash/development/base_component.py]: Either a `\n` or whitespace after the end of the sentence

Comment (alexcjohnson) [reply dash/development/base_component.py]: good call -> https://github.com/plotly/dash/pull/1026/commits/4d96e0dbf3c96ec19683c4e68c9e350302cb6da8

**Comment by Marc-Andre-Rivet** [comment]
Diff in https://percy.io/plotly/dash/builds/3264807/view/203019307/1280?mode=diff&browser=firefox&snapshot=203019307 looks related to async / the behavior is fine but without `eager_loading=True` the graph displays locally with a noticeable delay, after the `1` appears on the devtools icon -- the screenshot occurs before it renders.

**Comment by ohaibbq** [comment]
Hey y'all,

We're running into this exception while trying to use multiple props as `PropType.node`. [React's docs suggest](https://reactjs.org/docs/composition-vs-inheritance.html#containment) that if you have multiple pieces of content to dynamically add to a component, you should use your own props scheme rather than `props.children`. i.e. `props.leftChildren` and `props.rightChildren`.

Was this diff supposed to intentionally disallow this case? If not, I think we should add a `PropType` check to ensure that the prop type for `k` is not a `PropType.node` before raising this error.

**Pull request merged**